compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -80,6 +80,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let Some((def_id_or_name, output, inputs)) = self.extract_callable_info(expr, found)
             else { return false; };
         if can_satisfy(output) {
+            // suggest `unwrap_or_else` for `Option` and `Result`, when a closure is given
```

Comment:
Please move this logic to a different function. It doesn't make sense for this to live in `suggest_fn_call` -- since this isn't suggesting a fn call.

Reply:
Moved to function `suggest_else_fn_with_closure`.